src/ansi.rs

```diff
@@ -0,0 +1,1814 @@
+            StandardCharset::SpecialCharacterAndLineDrawing => match c {
+                '`' => '◆',
+                'a' => '▒',
+                'b' => '\t',
+                'c' => '\u{000c}',
+                'd' => '\r',
+                'e' => '\n',
+                'f' => '°',
+                'g' => '±',
+                'h' => '\u{2424}',
+                'i' => '\u{000b}',
+                'j' => '┘',
+                'k' => '┐',
+                'l' => '┌',
+                'm' => '└',
+                'n' => '┼',
+                'o' => '⎺',
+                'p' => '⎻',
+                'q' => '─',
+                'r' => '⎼',
+                's' => '⎽',
+                't' => '├',
+                'u' => '┤',
+                'v' => '┴',
+                'w' => '┬',
+                'x' => '│',
+                'y' => '≤',
+                'z' => '≥',
+                '{' => 'π',
+                '|' => '≠',
+                '}' => '£',
+                '~' => '·',
+                _ => c,
+            },
+        }
+    }
+}
+
+impl<'a, H, W> Perform for Performer<'a, H, W>
+where
+    H: Handler<W> + 'a,
+    W: 'a,
+{
+    #[inline]
+    fn print(&mut self, c: char) {
+        self.handler.input(c);
+        self.state.preceding_char = Some(c);
+    }
+
+    #[inline]
+    fn execute(&mut self, byte: u8) {
+        match byte {
+            C0::HT => self.handler.put_tab(1),
+            C0::BS => self.handler.backspace(),
+            C0::CR => self.handler.carriage_return(),
+            C0::LF | C0::VT | C0::FF => self.handler.linefeed(),
+            C0::BEL => self.handler.bell(),
+            C0::SUB => self.handler.substitute(),
+            C0::SI => self.handler.set_active_charset(CharsetIndex::G0),
+            C0::SO => self.handler.set_active_charset(CharsetIndex::G1),
+            _ => debug!("[unhandled] execute byte={:02x}", byte),
+        }
+    }
+
+    #[inline]
+    fn hook(&mut self, params: &Params, intermediates: &[u8], ignore: bool, action: char) {
+        match (action, intermediates) {
+            ('s', [b'=']) => {
+                // Start a synchronized update. The end is handled with a separate parser.
+                if params.iter().next().map_or(false, |param| param[0] == 1) {
+                    self.state.dcs = Some(Dcs::SyncStart);
+                }
+            },
+            _ => debug!(
+                "[unhandled hook] params={:?}, ints: {:?}, ignore: {:?}, action: {:?}",
+                params, intermediates, ignore, action
+            ),
+        }
+    }
+
+    #[inline]
+    fn put(&mut self, byte: u8) {
+        debug!("[unhandled put] byte={:?}", byte);
+    }
+
+    #[inline]
+    fn unhook(&mut self) {
+        match self.state.dcs {
+            Some(Dcs::SyncStart) => {
+                #[cfg(not(feature = "no_std"))]
+                {
+                    self.state.sync_state.timeout = Some(Instant::now() + SYNC_UPDATE_TIMEOUT);
+                }
```

Comment:
I'd just like to point out that this completely breaks this crate when it is excluded. Just randomly leaving out arbitrary statements when they don't work with a specific feature isn't really a good way to go about things.

Reply:
Unfortunately, no_std does not have `Instant` struct nor require the developer to implement any means to get current time. But I can create `TimeProvider` trait which returns the current time in milliseconds as `u64` and add it to `Processor` as an argument. What do you think, is this solution suitable?